tt_eager/tt_dnn/op_library/backward/backward_ops.cpp

```diff
@@ -2,124 +2,187 @@
-        power_input = recip(power_input,output_mem_config);
+    if (exponent < 1.0f) {
+        power_input = recip(power_input, output_mem_config);
     }
 
     Tensor result = mul_unary(power_input, exponent, output_mem_config);
     Tensor final_result = mul(result, grad, std::nullopt, output_mem_config);
-    final_result = where(gte_unary(final_result, 3.4e+38, output_mem_config), std::numeric_limits<float>::infinity(), where(lte_unary(final_result, -3.4e+38, output_mem_config), -std::numeric_limits<float>::infinity(), final_result, output_mem_config), output_mem_config);
+    final_result = where(
+        gte_unary(final_result, 3.4e+38, output_mem_config),
+        std::numeric_limits<float>::infinity(),
+        where(
+            lte_unary(final_result, -3.4e+38, output_mem_config),
+            -std::numeric_limits<float>::infinity(),
+            final_result,
+            output_mem_config),
+        output_mem_config);
     grad_tensor.emplace_back(final_result);
     return grad_tensor;
 }
-std::vector<Tensor> unary_pow_bw(const Tensor& grad, const Tensor& input, float exponent, const MemoryConfig& output_mem_config)
-{
+std::vector<Tensor> unary_pow_bw(
+    const Tensor& grad, const Tensor& input, float exponent, const MemoryConfig& output_mem_config) {
     return operation::decorate_as_composite(__func__, _unary_pow_bw)(grad, input, exponent, output_mem_config);
 }
 
-std::vector<Tensor> _unary_add_bw(const Tensor& grad, const Tensor& input, float alpha, const MemoryConfig& output_mem_config) {
+std::vector<Tensor> _unary_add_bw(
+    const Tensor& grad, const Tensor& input, float alpha, const MemoryConfig& output_mem_config) {
     std::vector<Tensor> grad_tensor;
     grad_tensor.emplace_back(grad);
     return grad_tensor;
 }
-std::vector<Tensor> unary_add_bw(const Tensor& grad, const Tensor& input, float alpha, const MemoryConfig& output_mem_config)
-{
+std::vector<Tensor> unary_add_bw(
+    const Tensor& grad, const Tensor& input, float alpha, const MemoryConfig& output_mem_config) {
     return operation::decorate_as_composite(__func__, _unary_add_bw)(grad, input, alpha, output_mem_config);
 }
 
-
-std::vector<Tensor> _mul_bw(const Tensor& grad, const Tensor& input_a, const Tensor& input_b, const MemoryConfig& output_mem_config) {
-    std::vector<Tensor> grad_tensor;
-    Tensor grad_a = mul(grad, input_b, std::nullopt, output_mem_config);
-    grad_tensor.emplace_back(grad_a);
-    Tensor grad_b = mul(grad, input_a, std::nullopt, output_mem_config);
-    grad_tensor.emplace_back(grad_b);
-    return grad_tensor;
+std::vector<std::optional<Tensor>> _mul_bw(
+    const Tensor& grad,
+    const Tensor& input_a,
+    const Tensor& input_b,
+    const MemoryConfig& output_mem_config,
+    const std::vector<bool>& are_required_outputs,
+    std::vector<std::optional<Tensor>> output_tensor) {
+    if (are_required_outputs.at(0)) {
+        output_tensor.at(0) = mul(grad, input_b, std::nullopt, output_mem_config);
```

Comment:
As mentioned https://github.com/tenstorrent/tt-metal/pull/8595#discussion_r1612773789,  the current implementation results in the `output_tensor` holding a new Tensor object returned from `mul`, which is not the intended behavior. Instead, the user-provided `output_tensor` should be directly passed to the `mul` function to ensure it is used and modified internally.

Reply:
ok I understand what you mean. once this PR for eltwise binary ops  is merged https://github.com/tenstorrent/tt-metal/pull/8394 we'll be able to pass output_tensor to those binary ops used within composite. 
CC @eyonland

Reply:
That's correct.  Thanks @dongjin-na